lib/pbench/server/cache_manager.py

```diff
@@ -82,6 +93,41 @@ def __str__(self) -> str:
         return f"An error occurred while changing file permissions of {self.tarball}: {self.error}"
 
 
+class FileInfo:
+    def __init__(self, dir_path: Path, path: Path):
+        """Collects the file info
+
+        Args:
+            dir_path: root directory parent path
+            path: path to a file/directory
+        """
+        self.name = path.name
+        self.location = path.relative_to(dir_path)
+        self.resolve_path = None
+        self.resolve_type = None
+        self.size = None
+        if path.is_symlink():
+            self.type = "symlink"
+            try:
+                resolve_path = path.resolve(strict=True)
+                self.resolve_type = "directory" if resolve_path.is_dir() else "file"
+            except FileNotFoundError:
+                resolve_path = path.readlink()
+                self.resolve_type = "symlink"
+
+            try:
+                self.resolve_path = resolve_path.relative_to(dir_path)
+            except ValueError:
+                self.resolve_path = resolve_path
```

Comment:
This is starting to look pretty good, but there are a couple of details here which we should touch up.

First, I think we should defer setting the `resolve_type` until the end (or, at least, reconsider it there):  if the referent path resolves to an existing file or directory which is outside the results area, I think we want the type to be set to `"symlink"`, not `"directory"` or `"file"` as it currently does at line 113 -- that is, we should treat the referent in the same way that we treat broken symlinks.  Second, I do think we should set the resolution type to `"other"` if the target is not a directory or regular file (or broken symlink).

So, I think I would code this something like this:
```python
        if path.is_symlink():
            self.type = "symlink"
            try:
                resolve_path = path.resolve(strict=True)
            except FileNotFoundError:
                resolve_path = path.readlink()
            try:
                self.resolve_path = resolve_path.relative_to(dir_path)
            except ValueError:
                self.resolve_path = resolve_path
                self.resolve_type = "symlink"
            else:
                resolve_path = dir_path / resolve_path
                if not resolve_path.exists():
                    self.resolve_type = "symlink"
                elif resolve_path.is_dir():
                    self.resolve_type = "directory"
                elif resolve_path.is_file():
                    self.resolve_type = "file"
                else:
                    self.resolve_type = "other"
```
That is:
- get the `resolve_path` either by resolving the path or, if that fails, by reading the symlink
- generate the path relative to the root of the results tree; if that fails, use the path value that we have and mark the resolution as a `"symlink"`
- if we were able to generate a relative path, reassemble the absolute path from the root of the results tree and the relative path:
  - if that does not exist, then the resolution type is a (broken) `"symlink"`; otherwise,
  - if that is a directory, then the resolution type is a `"directory"`; otherwise,
  - if that is a regular file, then the resolution type is a `"file"`; otherwise,
  - the resolution type is an `"other"`.